indexer/lib/advertisement-walker.js

```diff
@@ -151,7 +151,7 @@ export async function processNextAdvertisement ({
   assert(state.tail)
 
   try {
-    const { previousAdvertisementCid, ...entry } = await fetchAdvertisedPayload(
+    const { previousAdvertisementCid, entriesFetchError, ...entry } = await fetchAdvertisedPayload(
```

Comment:
why is `entriesFetchError` plural and `entry` singular?

Reply:
Let me explain how this works:

- First, we fetch the advertisement object. This object contains a CID linking to another object containing (payload block) entries.
- Second, we fetch the object with the entries.
- Finally, we take the first entry and discard the rest.

So, the error happens when we are fetching entries, but then we return only a single entry.